bug: XPRC ci not stable

### Current Behavior


https://github.com/apache/apisix/actions/runs/5505200613/jobs/10057126711?pr=9580


```log
Test Summary Report
-------------------
t/xds-library/config_xds_2.t                  (Wstat: 0 Tests: 35 Failed: 10)
  Failed tests:  15-24
  Parse errors: No plan found in TAP output
t/xrpc/redis.t                                (Wstat: 256 Tests: 54 Failed: 1)
  Failed test:  19
  Non-zero exit status: 1
t/xrpc/redis2.t                               (Wstat: 256 Tests: 13 Failed: 1)
  Failed test:  11
  Non-zero exit status: 1
Files=115, Tests=3111, 732 wallclock secs ( 0.69 usr  0.13 sys + 131.35 cusr 38.75 csys = 170.92 CPU)
Result: FAIL
Result: FAIL
Rerun t/xds-library/config_xds_2.t t/xrpc/redis.t t/xrpc/redis2.t
#   Failed test 't/xrpc/redis2.t TEST 3: verify the data received by the log server - grep_error_log_out (req 0)'
#   at /home/runner/work/apisix/apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1220.
#                   ''
#     doesn't match '(?^:message received:.*\"redis_cmd_line\":\"hmset animals dog bark cat meow\")'
# Looks like you failed 1 test of 13.
++ rerun_flaky_tests /tmp/test.result
++ tail -1 /tmp/test.result
++ grep 'Result: PASS'
++ tail -1 /tmp/test.result
++ grep 'Result: FAIL'
++ local tests
++ local n_test
+++ awk '/^t\/.*.t\s+\(.+ Failed: .+\)/{ print $1 }' /tmp/test.result
++ tests='t/xds-library/config_xds_2.t
t/xrpc/redis.t
t/xrpc/redis2.t'
+++ echo 't/xds-library/config_xds_2.t
t/xrpc/redis.t
t/xrpc/redis2.t'
+++ wc -l
++ n_test=3
++ '[' 3 -gt 10 ']'
+++ echo 't/xds-library/config_xds_2.t
t/xrpc/redis.t
t/xrpc/redis2.t'
+++ xargs
++ echo 'Rerun t/xds-library/config_xds_2.t t/xrpc/redis.t t/xrpc/redis2.t'
+++ echo 't/xds-library/config_xds_2.t
t/xrpc/redis.t
t/xrpc/redis2.t'
+++ xargs
++ FLUSH_ETCD=1
++ prove --timer -I./test-nginx/lib -I./ t/xds-library/config_xds_2.t t/xrpc/redis.t t/xrpc/redis2.t
[05:41:54] t/xds-library/config_xds_2.t .. ok    14329 ms ( 0.01 usr  0.00 sys +  0.88 cusr  0.20 csys =  1.09 CPU)
[05:42:02] t/xrpc/redis.t ................ ok     8096 ms ( 0.00 usr  0.00 sys +  2.29 cusr  0.83 csys =  3.12 CPU)

#   Failed test 't/xrpc/redis2.t TEST 3: verify the data received by the log server - grep_error_log_out (req 0)'
#   at /home/runner/work/apisix/apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1220.
#                   ''
#     doesn't match '(?^:message received:.*\"redis_cmd_line\":\"hmset animals dog bark cat meow\")'
# Looks like you failed 1 test of 13.
[05:42:05] t/xrpc/redis2.t ............... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/13 subtests 
[05:42:05]

Test Summary Report
-------------------
t/xrpc/redis2.t             (Wstat: 256 Tests: 13 Failed: 1)
  Failed test:  11
  Non-zero exit status: 1
Files=3, Tests=93, 25 wallclock secs ( 0.03 usr  0.01 sys +  3.81 cusr  1.18 csys =  5.03 CPU)
Result: FAIL
Error: Process completed with exit code 1.
```

### Expected Behavior

pass ci

### Error Logs

https://github.com/apache/apisix/actions/runs/5505200613/jobs/10057126711?pr=9580

### Steps to Reproduce

https://github.com/apache/apisix/actions/runs/5505200613/jobs/10057126711?pr=9580

### Environment

- APISIX version (run `apisix version`): master 
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

--- a/t/xrpc/redis2.t
+++ b/t/xrpc/redis2.t
@@ -16,6 +16,8 @@
 #
 use t::APISIX;
 
+log_level("warn");
+
 my $nginx_binary = $ENV{'TEST_NGINX_BINARY'} || 'nginx';
 my $version = eval { `$nginx_binary -V 2>&1` };
 
@@ -117,7 +119,7 @@ passed
                             {
                                 name = "syslog",
                                 filter = {
-                                    {"rpc_time", ">=", 0.01},
+                                    {"rpc_time", ">=", 0.001},
                                 },
                                 conf = {
                                     host = "127.0.0.1",
@@ -149,6 +151,7 @@ passed
 
 
 === TEST 3: verify the data received by the log server
+--- stream_conf_enable
 --- config
     location /t {
         content_by_lua_block {
@@ -190,11 +193,10 @@ passed
 hmset animals: OK
 hmget animals: barkmeow
 ping: pong
---- stream_conf_enable
 --- wait: 1
 --- grep_error_log eval
-qr/message received:.*\"redis_cmd_line\":[^,]+/
+qr/message received:.*\"redis_cmd_line\":[^}|^,]+/
 --- grep_error_log_out eval
-[qr/message received:.*\"redis_cmd_line\":\"hmset animals dog bark cat meow\"/,
-qr/message received:.*\"redis_cmd_line\":\"hmget animals dog cat\"/,
-qr/message received:.*\"redis_cmd_line\":\"ping\"/]
+qr{message received:.*\"redis_cmd_line\":\"hmset animals dog bark cat meow\"(?s).*
+message received:.*\"redis_cmd_line\":\"hmget animals dog cat\"(?s).*
+message received:.*\"redis_cmd_line\":\"ping\"}

